org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/presentation/DSPDebugModelPresentation.java

```diff
@@ -72,6 +72,26 @@ public String getText(Object element) {
 		} else {
 			// Use default TODO should the entire default be copied here?
 			label.append(DebugUIPlugin.getDefaultLabelProvider().getText(element));
+
+			if (element instanceof DSPStackFrame frame) {
+				try {
+					final int line = frame.getLineNumber();
+					if (line > 0) {
+						final String source = frame.getSourceName();
```

Comment:
`org.eclipse.lsp4j.debug.StackFrame.getSource()` is optional, so I think we need to check for `null`. Don't you think so?

Reply:
wow, you have eagle eyes! this was actually a bug in DSPStackFrame.